### PR TITLE
Remove STANDARD_LOCKTIME_VERIFY_FLAGS and mempool policy's flags

### PR DESCRIPTION
**Pull request by NicolasDorier**
Remove flags from CheckFinalTx and removeForReorg.
MTP is always enforced since 0.11.2 in mempool.
See release notes: https://bitcoin.org/en/release/v0.11.2

### CODE REVIEW

**Comment by laanwj** [comment]
Needs rebase. I agree moving the `CTxMemPool::removeForReorg` function seems unnecessary.

**Comment by jtimon** [comment]
+1 nack on moving CTxMemPool::removeForReorg for no apparent reason.
I have mixed feelings about this. On one hand, I kind of hate default values for params (like flags=-1) and this has some nice cleanups. On the other hand, I would like to be more explicit about flags everywhere, not less.
I've been saying this multiple times in "random" places, but...can't we just move LOCKTIME_VERIFY_SEQUENCE and LOCKTIME_MEDIAN_TIME_PAST to script/interpreter.h?
Never mind, don't want to hijack the PR, I will just open another related one.

**Comment by NicolasDorier** [comment]
rebased, @jtimon I'm not sure what you mean by "moving removeForReorg" ? can you check again ?

**Comment by btcdrak** [comment]
@NicolasDorier Looks much better now.

**Comment by NicolasDorier** [comment]
Travis on Compiler: No Wallet "fatal: reference is not a tree: f64477fe5a549cd54eebcbc68b679a397415d43b". Travis problem ?

image

File: src/txmempool.cpp [hunk 1]
```diff
@@ -528,7 +528,7 @@ void CTxMemPool::removeForReorg(const CCoinsViewCache *pcoins, unsigned int nMem
         const CTransaction& tx = it->GetTx();
         LockPoints lp = it->GetLockPoints();
         bool validLP =  TestLockPointValidity(&lp);
-        if (!CheckFinalTx(tx, flags) || !CheckSequenceLocks(tx, flags, &lp, validLP)) {
+        if (!CheckFinalTx(tx, flags) || !CheckSequenceLocks(tx, LOCKTIME_VERIFY_SEQUENCE, &lp, validLP)) {
```
Comment (MarcoFalke) [inline src/txmempool.cpp]:
```cpp
CheckFinalTx(tx, flags)
```

```
txmempool.cpp:531:31: error: ‘flags’ was not declared in this scope

**Comment by jtimon** [comment]
@NicolasDorier I meant what @laanwj (and I would swear someone else before him, but I can't see the "Someone commented on an outdated diff" thing) said (that CTxMemPool::removeForReorg was being moved unnnecessarly [and probably unintentionally too]). Looks fixed now.

Rewarding the only-slightly related comment, this is what I mean is in #7779 .

File: src/txmempool.cpp (hunk 1 above)
Comment (NicolasDorier) [inline src/txmempool.cpp]: I'm not using flags anymore as you can see, I think you checkout gone wrong ?

**Comment by NicolasDorier** [comment]
@jtimon I think the problem was the way github presented the diff. I never intended to move a function. Somehow the rebase fixed it.

**Comment by jtimon** [comment]
Oh, I was complaining about github's traceability, but maybe we don't want to keep traceability of their bugs, only ours (not saying that accidentally moving a function was a bug if that was the case). Anyway, mystery solved or not, let's move on: it's fixed.

File: src/main.cpp [hunk 2]
```diff
@@ -681,18 +681,10 @@ bool IsFinalTx(const CTransaction &tx, int nBlockHeight, int64_t nBlockTime)
     return true;
 }
 
-bool CheckFinalTx(const CTransaction &tx, int flags)
+bool CheckFinalTx(const CTransaction &tx)
 {
     AssertLockHeld(cs_main);
 
-    // By convention a negative value for flags indicates that the
```
Comment (jtimon) [inline src/main.cpp]: I always dislike removing documentation but I was never sure where to move this or how to replace it.
Is https://github.com/bitcoin/bitcoin/pull/7779/commits/22a80055021cdfa6f6bd1496ce15308285424c29 useful?

Comment (NicolasDorier) [inline src/main.cpp]: I like having a single flag, not sure about the pratical purpose of VALIDATION_LEVEL though,but I think would be better suited for a separate PR, this one is more about removing useless code only than refactoring.

File: src/policy/policy.h [hunk 3]
```diff
@@ -46,10 +46,6 @@ static const unsigned int STANDARD_SCRIPT_VERIFY_FLAGS = MANDATORY_SCRIPT_VERIFY
 /** For convenience, standard but not mandatory verify flags. */
 static const unsigned int STANDARD_NOT_MANDATORY_VERIFY_FLAGS = STANDARD_SCRIPT_VERIFY_FLAGS & ~MANDATORY_SCRIPT_VERIFY_FLAGS;
 
-/** Used as the flags parameter to sequence and nLocktime checks in non-consensus code. */
-static const unsigned int STANDARD_LOCKTIME_VERIFY_FLAGS = LOCKTIME_VERIFY_SEQUENCE |
```
Comment (jtimon) [inline src/policy/policy.h]: put in STANDARD_NOT_MANDATORY_VERIFY_FLAGS until putting it in mandatory? ping @TheBlueMatt

**Comment by morcos** [comment]
I'm in favor of cleaning up these flags, but I'm not sure this is the right approach.

Probably something along the lines of what @jtimon is suggesting where there is just one set of consensus flags that are used would be better.  

I think we still want to pass flags to CheckFinalTx because we might want to preserve a way of accepting non-MTP locktime txs in the mempool.  Certainly its a bit premature to eliminate that until the soft forks activate.

**Comment by NicolasDorier** [comment]
Closing for jtimon approach
